src/languages/en.ts

```diff
@@ -2340,6 +2340,17 @@ export default {
                     }
                 }
             },
+            syncError: (integration?: ConnectionName): string => {
+                switch (integration) {
+                    case CONST.POLICY.CONNECTIONS.NAME.QBO:
+                        return "Couldn't connect to QuickBooks Online due to incorrect credentials.";
+                    case CONST.POLICY.CONNECTIONS.NAME.XERO:
+                        return "Couldn't connect to Xero due to incorrect credentials.";
+                    default: {
+                        return "Couldn't connect to integration due to incorrect credentials.";
```

Comment:
I don't know if "due to incorrect credentials" is the correct thing to say here. The `lastSync.isSuccessful` flag is used for any sync error, not just invalid credentials. It could be something like third-party was down, or the user is missing permissions in the third-party for what we're trying to do, etc.

Reply:
Thank you François for answering quickly my previous question and raising this point!

cc @trjExpensify @zanyrenney in the Xero DD, and in the Figma we would show this `Couldn't connect to Xero due to incorrect credentials.` messageN. But since it can be a lot of completely different things, maybe we need to change it?

Before having this discussion, I suggested to @SzymczakJ using as copy something like `Couldn't connect to {integrationName}.`. Generic yes, but it fits all the possible use cases. What do you think?

Reply:
Zany might have more context, but I think this was just an example where incorrect credentials are at fault, but the intention is to pass in the most relevant error we have. Do we not have all of these in OldDot already?

Reply:
We do not have something to identify the specific error when syncing, we only have these properties:

```
{
  "lastSync": {
    "successfulDate": "2024-03-25T10:32:18+0000",
    "isSuccessful": true,
    "errorDate": "2024-03-22T10:20:49+0000",
    "source": "EXPENSIFYWEB"
},
```
From my understanding, since the sync issue can be a myriad of things, we're not able to precisely identify which error provoked that.

Currently, we show this if there's a sync error:
![image](https://github.com/Expensify/App/assets/21549771/262febc7-7df5-4d34-878d-20d15e675db6)

Reply:
Okay cool, so in this case for NewDot, would the button inside the three dot overflow menu read `Enter credentials` in this scenario? So our direction would be to point to that instead of a "Reconnect" button?

Reply:
I.e 

`genericConnectFailedMessage: 'Your connection to <%- connectionName %> has expired. You can correct this by clicking <b>Enter credentials</b> and reauthorizing Expensify with <%- connectionName %>.',`

Reply:
Showing it in the three dots overflow menu sounds good. The wording `Enter credentials` might be confusing though as we don't always have to enter credentials, here is what is shown when I click on reconnect because I have expired credentials:
![image](https://github.com/Expensify/App/assets/21549771/496b03ad-7d1e-4f21-a84c-a34e867eb3cc)

Reply:
Ah, wow.. interesting you don't have to enter credentials again if they've expired. Okay, so I guess in this case we would have a `Reconnect` button in the overflow menu and keep the existing copy?

Reply:
FYI this is all gonna be handled in the errors doc. Commented about that [here]. Not sure what we should do with this PR, but I'm leaning towards closing it honestly.

Reply:
[Looks like](https://github.com/Expensify/App/issues/42250#issuecomment-2123609669) we might want to keep it, at least for errors when connecting (not autoSync/export/manual export errors that are out of scope). Especially to demo it at XeroCon.
Thoughts @trjExpensify @arosiclair?

Reply:
Yeah I think we can keep it though the error we show here should just a generic message like Francois mentioned

Reply:
So I should proceed with generic error message like "Couldn't connect to Xero due to incorrect credentials.", right? @trjExpensify

Reply:
Yep! This one I believe:

>genericConnectFailedTitle: 'Couldn\'t connect to <%- connectionName %>',